### PR TITLE
Add others's status in hpa describe

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
Add other status in HPA describe

- rs
- deployment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add all kinds of resource objects' statuses in HPA description.
```

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
/assign @liggitt 
could you take a look?

**Comment by ericchiang** [comment]
Can you add a description of what this does? Why is it needed?

**Comment by zhangxiaoyu-zidif** [comment]
@ericchiang 
hi, I see `The Horizontal Pod Autoscaler automatically scales the number of pods in a replication controller, deployment or replica set based on observed CPU utilization` in https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/ and the code only contains rc's description, I guess we may need add rs and deploy's description on HPA

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -2875,6 +2875,24 @@ func describeHorizontalPodAutoscaler(hpa *autoscaling.HorizontalPodAutoscaler, e
 				w.Write(LEVEL_0, "failed to check Replication Controller\n")
 			}
 		}
+		if strings.ToLower(hpa.Spec.ScaleTargetRef.Kind) == "replicaSet" {
```
Comment (liggitt) [inline pkg/printers/internalversion/describe.go]: ToLower will never match "replicaSet"

We should not be adding hard coded scale targets to this function. If we want to use the generic scale client to fetch scale status, that would be better

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -2875,6 +2875,24 @@ func describeHorizontalPodAutoscaler(hpa *autoscaling.HorizontalPodAutoscaler, e
...
+			w.Write(LEVEL_0, "ReplicaSet pods:\t")
+			rs, err := d.client.Extensions().ReplicaSets(hpa.Namespace).Get(hpa.Spec.ScaleTargetRef.Name, metav1.GetOptions{})
+			if err == nil {
+				w.Write(LEVEL_0, "%d current / %d desired\n", rs.Status.Replicas, rs.Spec.Replicas)
+			} else {
+				w.Write(LEVEL_0, "failed to check ReplicaSet\n")
+			}
+		}
+		if strings.ToLower(hpa.Spec.ScaleTargetRef.Kind) == "deployment" {
```
Comment (liggitt) [inline pkg/printers/internalversion/describe.go]: Same here. Don't add hardcoded scale targets. Replace the existing one with a call using the polymorphic scale client

Comment (zhangxiaoyu-zidif) [reply pkg/printers/internalversion/describe.go]: I find we can use subsource to show the current replicas and desired replicas, so I use it instead. =)

**Comment by zhangxiaoyu-zidif** [comment]
@liggitt  PTAL

**Pull request merged**